### PR TITLE
fix: prevent info from running unnecessarily

### PR DESCRIPTION
**Pull request by jamesgeorge007**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
Yup

**If relevant, did you update the documentation?**
Nope

**Summary**
- Make use of `commander.js` to parse the available commands.
- Prevents `info` from running unnecessarily.

**Does this PR introduce a breaking change?**
Nope

**Other information**
N/A

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
`webpack-cli --help` shows up the custom help information since we're overriding the default behavior.

```bash

          ⬡                     webpack                     ⬡

		         https://webpack.js.org

		The build tool for modern web applications

		Usage: `webpack [...options] | <command>`

		Example: `webpack help --flag | <command>`



Available Commands

  init | c      Initialize a new webpack configuration                 
  migrate | m   Migrate a configuration to a new version               
  loader | l    Scaffold a loader repository                           
  plugin | p    Scaffold a plugin repository                           
  info | i      Outputs information about your system and dependencies 
  serve | s     Run the webpack Dev Server                             

Options

  --entry string[]         The entry point(s) of your application e.g.          
                           ./src/main.js                                        
  -c, --config string      Provide path to a webpack configuration file e.g.    
                           ./webpack.config.js                                  
  -m, --merge string       Merge a configuration file using webpack-merge e.g.  
                           ./webpack.config.js                                  
  --progress               Print compilation progress during build              
  --silent                 Disable any output that webpack makes                
  --help                   Outputs list of supported flags                      
  -o, --output string      Output location of the file generated by webpack     
                           e.g. ./dist/                                         
  -t, --target string      Sets the build target e.g. node                      
  -w, --watch              Watch for files changes                              
  -h, --hot                Enables Hot Module Replacement                       
  --no-hot                 Disables Hot Module Replacement                      
  -s, --sourcemap string   Determine source maps to use                         
  --prefetch string        Prefetch this request                                
  -j, --json               Prints result as JSON                                
  -d, --dev                Run development build                                
  -p, --prod               Run production build                                 
  --mode string            Defines the mode to pass to webpack                  
  -v, --version            Get current version                                  
  --node-args string[]     NodeJS flags                                         
  --stats string           It instructs webpack on how to treat the stats e.g.  
                           verbose                                              
  --verbose                It tells webpack to output all the information       
  --env string             Environment passed to the configuration when it is a 
                           function                                             

                  Made with ♥️  by the webpack team
```

**Comment by jamesgeorge007** [comment]
Let's revisit the refactoring part in a future PR :+1:

File: packages/webpack-cli/lib/bootstrap.js [hunk 1]
```diff
@@ -38,74 +30,72 @@ async function runCLI(cli, commandIsUsed) {
...
+            const newArgKeys = Object.keys(argsMap).filter((arg) => !keysToDelete.includes(argsMap[arg].pos));
+            // eslint-disable-next-line require-atomic-updates
+            process.argv = newArgKeys;
+            args = argParser('', core, process.argv);
+            await cli.run(args.opts, core);
+            process.stdout.write('\n');
+            logger.warn('Duplicate flags found, defaulting to last set value');
+        } else {
+            logger.error(err);
+            return;
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/bootstrap.js:98]: I am afraid for this place, we don't set `exitCode` here, so errors will have `exitCode: 0`, also `eslint-loader`/etc will be broken, can you investigate it and add couple tests in next PRs?

Comment (jamesgeorge007) [reply packages/webpack-cli/lib/bootstrap.js:98]: Sure

**Pull request merged**